lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -927,6 +927,60 @@ OpFoldResult AtenViewOp::fold(FoldAdaptor adaptor) {
   return getOperand(0);
 }
 
+void AtenViewOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
+                                             MLIRContext *context) {
+  // Canonicalize `aten.view` to `aten.unsqueeze`
+  patterns.add(+[](AtenViewOp op, PatternRewriter &rewriter) {
+    Location loc = op->getLoc();
+    auto inputType = op.getSelf().getType().dyn_cast<BaseTensorType>();
+    if (!inputType || !inputType.hasSizes())
+      return failure();
+    ArrayRef<int64_t> inputShape = inputType.getSizes();
+    int64_t inputRank = inputShape.size();
+
+    auto resType = op.getType().dyn_cast<BaseTensorType>();
+    if (!resType || !resType.hasSizes())
+      return failure();
+    ArrayRef<int64_t> resShape = resType.getSizes();
+    int64_t resRank = resShape.size();
+    if (inputRank >= resRank)
+      return failure();
+    SmallVector<int64_t> dimList;
+    int64_t inputPtr = 0;
```

Comment:
[?,?] -> [?,?,1]  isn't necessarily an unsqueeze (it could be [2,3] -> [3,2,1]). Possible easy fix: check for static shapes.

Reply:
This is a question for @Shukla-Gaurav but we encountered a bunch of problems with constant propagation and i'm guessting it has something to do with that.

Reply:
This is a cumulative patch for making the opt model work. There was an issue with this particular case of view of in the opt model, so added the canonicalizer as a temporary fix. Will raise a separate patch if it is at all necessary.

Reply:
Ok. I'm just wondering if it's needed, because I don't seem to be hitting the view lowering issue with the latest torch-mlir.